### PR TITLE
Direct logout

### PR DESCRIPTION
**Pull request by louischan-oursky**
ref #1129

### CODE REVIEW

File: pkg/lib/oauth/oidc/handler/handler_end_session.go [hunk 1]
```diff
@@ -16,13 +16,41 @@ type WebAppURLsProvider interface {
...
+	Endpoints      oidc.EndpointsProvider
+	URLs           WebAppURLsProvider
+	IDTokens       IDTokenIssuer
+	SessionManager LogoutSessionManager
 }
 
 func (h *EndSessionHandler) Handle(s session.Session, req protocol.EndSessionRequest, r *http.Request, rw http.ResponseWriter) error {
+	idTokenHint := req.IDTokenHint()
+	if s != nil && idTokenHint != "" {
+		err := h.IDTokens.ValidateIDToken(idTokenHint)
```
Comment (carmenlau) [inline pkg/lib/oauth/oidc/handler/handler_end_session.go:38]: Seems we only validate the id token here, but we have not compared it with the current session?
If this is the case, there will be a problem that people can use another peoples'  id token to logout current user.

File: pkg/lib/oauth/oidc/id_token.go [hunk 2]
```diff
@@ -48,11 +94,16 @@ func (ti *IDTokenIssuer) IssueIDToken(client *config.OAuthClientConfig, s sessio
 	now := ti.Clock.NowUTC()
 
 	_ = claims.Set(jwt.AudienceKey, client.ClientID)
+
+	_ = claims.Set("sid", EncodeSID(s))
```
Comment (carmenlau) [inline pkg/lib/oauth/oidc/id_token.go:167]: Seems `sid` is custom claim. Will we don't some prefix to it, do we have similar case in the existing code base?

https://openid.net/specs/openid-connect-core-1_0.html#IDToken

Comment (louischan-oursky) [reply pkg/lib/oauth/oidc/id_token.go:167]: No. It is defined here. https://openid.net/specs/openid-connect-frontchannel-1_0.html#OPLogout

File: pkg/lib/oauth/oidc/handler/handler_end_session.go (hunk 1 above)
Comment (louischan-oursky) [reply pkg/lib/oauth/oidc/handler/handler_end_session.go:38]: This is quite tricky. Can we have a chat on this.

**Pull request merged**